src/log.js

```diff
@@ -144,6 +144,19 @@ class Log extends GSet {
     return Log.findTailHashes(this.values)
   }
 
+  /**
+   * Set a function to be called for every entry in a log that is about to
+   * be joined. The arguments will be the entry to verify and a reference
+   * to the log itself and the verification function should return true
+   * if the entry is valid and false if it is not. Logs containing entries
+   * marked as invalid by this function will not be joined
+   * 
+   * @memberof Log
+   */
+  set customVerificationFunction (verifyEntry) {
```

Comment:
Let's pass in the verify function through constructor options as we do with other "dependencies"? This way the user doesn't need to do `log.customVerificationFunction = () ...` afterwards. I feel that's an easier (safer?) way to do it. What do you think?

Reply:
I feel we should do the latter, ie. pass it to Store and then have Store pass it to the log.

Reply:
I tried adding a constructor argument, but in the tests for examples many Logs are created using Log.fromEntry or other factory functions like that.

I'm not sure what the best approach is, do you think we should add this option to every factory function? I ended up keeping the setter as well as the constructor argument for now.

I also noticed there was some weird jsdoc syntax before the constructor, I changed it, let me know if that works

Reply:
@haadcode ping

Reply:
Hey sorry for the delay! 

You're absolutely right, we need to add the verification function passing to the factory functions too in order to make this work. Fortunately it's only three places (afaict):
- https://github.com/orbitdb/ipfs-log/blob/master/src/log.js#L446
- https://github.com/orbitdb/ipfs-log/blob/master/src/log.js#L463
- https://github.com/orbitdb/ipfs-log/blob/master/src/log.js#L480

I would add it as the last argument (like you did in the constructor). 

(And on that note, I feel we should generally pass the various "options" through an options object and not individual argument, but that's something for future, not for this PR)

I still think we should not add a setter but rather pass it in through function calls in order to avoid accidental usage/mutation (note that we don't use setter anywhere in the code).

What do you think?

Reply:
Sounds like the best solution, I'll update the PR today :+1: Thanks for pointing out the factory functions